src/Http/Http.Extensions/src/HttpResponseJsonExtensions.cs

```diff
@@ -82,9 +82,28 @@ public static partial class HttpResponseJsonExtensions
             options ??= ResolveSerializerOptions(response.HttpContext);
 
             response.ContentType = contentType ?? JsonConstants.JsonContentTypeWithCharset;
+            // if no user provided token, pass the RequestAborted token and ignore OperationCanceledException
+            if (!cancellationToken.CanBeCanceled)
+            {
+                return WriteAsJsonAsyncSlow<TValue>(response.Body, value, options, response.HttpContext.RequestAborted);
+            }
+
             return JsonSerializer.SerializeAsync<TValue>(response.Body, value, options, cancellationToken);
         }
 
+        private static async Task WriteAsJsonAsyncSlow<TValue>(
+            Stream body,
+            TValue value,
+            JsonSerializerOptions? options,
+            CancellationToken cancellationToken)
+        {
+            try
+            {
+                await JsonSerializer.SerializeAsync<TValue>(body, value, options, cancellationToken);
+            }
+            catch (OperationCanceledException) { }
```

Comment:
Oh wow, I missed that we would have to do this. FYI @JamesNK \ @Tratcher in case you have thoughts about this

Reply:
What is the reason to catch the OCE? Throwing it on cancellation is standard.

Reply:
Following our pattern in send file https://github.com/dotnet/aspnetcore/issues/35330#issuecomment-898570531

Reply:
We want to not throw if you didn't explicitly pass the token